From aacd0781476ea652030ded863083d32ba4b69d29 Mon Sep 17 00:00:00 2001
From: tug <tug@skyth-tek.com>
Date: Sat, 19 Aug 2017 10:52:08 +0800
Subject: [PATCH 10/13] fix spdif HD audio


Signed-off-by: tug <tug@skyth-tek.com>
---
 sound/soc/rockchip/rk_spdif.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/rockchip/rk_spdif.c b/sound/soc/rockchip/rk_spdif.c
index 12f5f52a..c86b252 100755
--- a/sound/soc/rockchip/rk_spdif.c
+++ b/sound/soc/rockchip/rk_spdif.c
@@ -136,7 +136,7 @@
 #define INTCR_SDBEIE_MASK		(0x1 << 4)
 
 /* size * width: 16*4 = 64 bytes */
-#define SPDIF_DMA_BURST_SIZE		(16)
+#define SPDIF_DMA_BURST_SIZE		(8)
 
 struct rockchip_spdif_info {
 	spinlock_t lock;/*lock parmeter setting.*/
-- 
1.7.9.5

